system/metrics/sinks/elastic/elasticSink.cpp

```diff
@@ -162,26 +149,157 @@ bool ElasticMetricSink::getIndexConfig(const IPropertyTree *pSettingsTree)
         return false;
     }
 
-    // Initialize standard suffixes
-    if (!pIndexConfigTree->getProp("@countMetricSuffix", countMetricSuffix))
-        countMetricSuffix.append("count");
+    intializeElasticClient();
 
-    if (!pSettingsTree->getProp("@gaugeMetricSuffix", gaugeMetricSuffix))
-        gaugeMetricSuffix.append("gauge");
+    if (!validateIndex())
+        return false;
+
+    return getDynamicMappingSuffixesFromIndex(pIndexConfigTree);
+}
 
-    if (!pSettingsTree->getProp("@histogramMetricSuffix", histogramMetricSuffix))
-        histogramMetricSuffix.append("histogram");
 
-    return true;
+bool ElasticMetricSink::getDynamicMappingSuffixesFromIndex(const IPropertyTree *pIndexConfigTree)
+{
+    StringBuffer countSuffixMappingName;
+    if (!pIndexConfigTree->getProp("@countSuffixMappingName", countSuffixMappingName))
+        countSuffixMappingName.append("hpcc_metrics_count_suffix");
+
+    StringBuffer gaugeSuffixMappingName;
+    if (!pIndexConfigTree->getProp("@gaugeSuffixMappingName", gaugeSuffixMappingName))
+        gaugeSuffixMappingName.append("hpcc_metrics_gauge_suffix");
+
+    StringBuffer histogramSuffixMappingName;
+    if (!pIndexConfigTree->getProp("@histogramSuffixMappingName", histogramSuffixMappingName))
+        histogramSuffixMappingName.append("hpcc_metrics_histogram_suffix");
+
+    std::string endpoint;
+    endpoint.append("/").append(indexName.str()).append("/_mapping");
+    httplib::Result res = pClient->Get(endpoint.c_str(), elasticHeaders);
+    if (res->status != 200)
+    {
+        WARNLOG("ElasticMetricSink: Unable to retrieve index mapping for index %s", indexName.str());
+        return false;
+    }
+
+    nlohmann::json data = nlohmann::json::parse(res->body);
+
+    auto indexConfig = data[indexName.str()];
+    if (indexConfig.is_null())
+    {
+        WARNLOG("ElasticMetricSink: Mapping section for Index '%s' does not exist", indexName.str());
```

Comment:
how does one recover from this? Provide useful information to the target audience.